src/reducers/index.js

```diff
@@ -5,6 +5,10 @@ const initialSubscription = {
 }
 
 const subscribed = (state = initialSubscription, action) => {
+  if(action.type === undefined) {
+    return false
+  }
+
   switch (action.type) {
     case 'SET_SUBSCRIBED':
       return true
```

Comment:
I believe I catch the 'RANDOM-123' case here and return state.subscribed which I pass in as `{subscribed: true/false}`